core/src/main/java/com/netflix/conductor/core/utils/Lock.java

```diff
@@ -0,0 +1,45 @@
+package com.netflix.conductor.core.utils;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Interface implemented by a distributed lock client.
+ *
+ * A typical usage:
+ *   if (acquireLock(workflowId, 5, TimeUnit.MILLISECONDS)) {
+ *      [load and execute workflow....]
+ *      ExecutionDAO.updateWorkflow(workflow);  //use optimistic locking
+ *   } finally {
+ *     releaseLock(workflowId)
+ *   }
+ *
+ */
+
+public interface Lock {
+
+    /**
+     * acquires a re-entrant lock on lockId, blocks indefinitely on lockId until it succeeds
+     * @param lockId resource to lock on
+     * @return
+     */
+    void acquireLock(String lockId);
+
+    /**
+     * acquires a re-entrant lock on lockId, blocks for timeToTry duration before giving up
+     * @param lockId resource to lock on
+     * @param timeToTry blocks up to timeToTry duration in attempt to acquire the lock
+     * @param unit time unit
+     * @return
+     */
+    Boolean acquireLock(String lockId, long timeToTry, TimeUnit unit);
```

Comment:
Can we have input param a container object(LockOptions) and return a LockItem object. 

Advantages are : 
1) If we want to take in a new parameter for acquiring Lock. we would not have overloaded methods with a lot of parameters. For example, if we want a Lock Implementation which supports how long a lease can be taken. 

2) If we have a LockItem object which has when the lock has been acquired and how much time is still remaining. 
He can try to reacquire the lock before proceeding.